parsl/executors/high_throughput/mpi_resource_management.py

```diff
@@ -0,0 +1,191 @@
+import logging
+import multiprocessing
+import os
+import pickle
+import queue
+import subprocess
+from enum import Enum
+from typing import Dict, List
+
+from parsl.multiprocessing import SpawnContext
+from parsl.serialize import (pack_res_spec_apply_message,
+                             unpack_res_spec_apply_message)
+
+logger = logging.getLogger(__name__)
+
+
```

Comment:
maybe the behaviour in this class belongs with the relevant provider in parsl.providers?

Reply:
Conceptually yes, in that we are figuring out resources on a per job basis and is very tied to batch-jobs. Right now, this functionality is usable only at by the manager, which is why I placed it here. Might be worth a chat.